src/raps/execute.ts

```diff
@@ -125,50 +130,45 @@ export const walletExecuteRap = async (
   type: RapTypes,
   parameters: RapSwapActionParameters<'swap' | 'crosschainSwap' | 'claimBridge'>
 ): Promise<{ nonce: number | undefined; errorMessage: string | null }> => {
-  const rap: Rap = await createSwapRapByType(type, parameters);
+  // NOTE: We don't care to track claimBridge raps
+  const rap =
+    type === 'claimBridge'
+      ? await createSwapRapByType(type, parameters)
+      : await performanceTracking.getState().executeFn({
+          fn: createSwapRapByType,
+          screen: Routes.SWAP,
+          operation: TimeToSignOperation.CreateRap,
+        })(type, parameters);
 
   const { actions } = rap;
   const rapName = getRapFullName(rap.actions);
   let nonce = parameters?.nonce;
   let errorMessage = null;
-  if (actions.length) {
-    const firstAction = actions[0];
-    const actionParams = {
-      action: firstAction,
-      wallet,
-      rap,
-      index: 0,
-      baseNonce: nonce,
-      rapName,
-      flashbots: parameters?.flashbots,
-      gasParams: parameters?.gasParams,
-      gasFeeParamsBySpeed: parameters?.gasFeeParamsBySpeed,
-    };
 
-    const { baseNonce, errorMessage: error, hash } = await executeAction(actionParams);
+  if (actions.length) {
```

Comment:
To be honest I'm not sure why we extract the first action out and then loop over the rest. Happy to revert this change, but it didn't make any sense to me why we had this here.

Also, fwiw I tested with swap & swap+unlock and both worked just fine.

Reply:
Reverted, this can be part of a follow-up PR and doesn't need to be in this PR.